tests/Fixtures/SubscriptionRepositorySpy.php

```diff
@@ -26,4 +26,14 @@ public function getSubscriptions(): array {
 		return $this->subscriptions;
 	}
 
+	public function findByConfirmationCode( string $confirmationCode )
+	{
+		foreach( $this->subscriptions as $subscription ) {
+			if ( $subscription->getConfirmationCode() === $confirmationCode ) {
+				return $subscription;
+			}
+		}
```

Comment:
Actually, I accidentally created a new "InMemorySubscriptionRepository" here, because i really need the result. Will see if I can use the DBAL repo instead or if InMemorySubscriptionRepository has to be revived.